### ISSUE TITLE
not able to retrieve paid invoices

### ISSUE THREAD

**Issue by buschauer** [paypal/PayPal-PHP-SDK#94]
Unable to retrieve paid invoices using  Invoice::get..

Error message is

Method PayPal\Api\Invoice::getPayments() does not exist

So the following code needs to be added to PayPal/Api/Invoice.php

    /**
     * Set Payments
     * A list of Payment resources
     *
     * @param \PayPal\Api\Payment $payments
     *
     * @return $this
     */
    public function setPayments($payments)
    {
        $this->payments = $payments;

        return $this;
    }

    /**
     * Get Payments
     * A list of Payment resources
     *
     * @return \PayPal\Api\Payment
     */
    public function getPayments()
    {
        return $this->payments;
    }

**Comment by sebdesign** [comment]
I have exactly the same bug. In the commit e6fb9685099b35d5a8d777109d0c520667a73960 the `getPayments()` method is replaced with `getPaymentDetails()`, but for some reason `getPayments()` is still being called.

**Comment by jaypatel512** [comment]
The reason why it was changed was because the schema currently shows the response object of invoice to have payment_details, and so getPaymentDetails(). Look for response object in 

https://developer.paypal.com/webapps/developer/docs/api/#retrieve-an-invoice

However, can you confirm if the json response you are receiving is having payment_details or payment object ? It could be backward compatible change, that got removed. I can bring that getter and point to getPaymentDetails(), to resolve backward incompatibility issue.

**Comment by sebdesign** [comment]
How can I dump the JSON response with this SDK?

**Comment by jaypatel512** [comment]
If you have logging enabled, just make it as [shown here](https://github.com/paypal/PayPal-PHP-SDK/blob/master/sample/bootstrap.php#L78), and you should be able to get the response data:

**Comment by sebdesign** [comment]
It seems the json returns a `payments` key instead of a `payment_details`. Can I provide you any more info about this?

````
{
  "id":"INV2-RV68-2EBS-DTW4-KYHW",
  "number":"0005",
  "status":"MARKED_AS_PAID",
  "merchant_info":{
    "email":"info-facilitator@sebdesign.eu",
    "business_name":"Alkyonis Apartments",
    "phone":{
      "country_code":"30",
      "national_number":"2377051441"
    },
    "website":"http://www.alkyonis-apartments.gr",
    "address":{
      "line1":"Alkyonis Apartments",
      "city":"Ammouliani",
      "state":"Chalkidiki",
      "postal_code":"63075",
      "country_code":"GR",
      "phone":{
        "country_code":"30",
        "national_number":"2377051441"
      }
    }
  },
  "billing_info":[
    {
      "email":"sebastien@stonewave.net",
      "first_name":"Sebastien",
      "last_name":"Nikolaou"
    }
  ],
  "items":[
    {
      "name":"Booking",
      "description":"\nCheck in: 29/09/2015\nCheck out: 30/09/2015\nAdults: 1\nRooms: 101",
      "quantity":1.0,
      "unit_price":{
        "currency":"EUR",
        "value":"5.00"
      }
    }
  ],
  "invoice_date":"2015-01-09 PST",
  "payment_term":{
    "term_type":"NET_30",
    "due_date":"2015-02-08 PST"
  },
  "tax_calculated_after_discount":false,
  "tax_inclusive":false,
  "total_amount":{
    "currency":"EUR",
    "value":"5.00"
  },
  "payments":[
    {
      "type":"EXTERNAL",
      "method":"BANK_TRANSFER",
      "date":"2015-01-09 23:59:59 PST"
    }
  ],
  "metadata":{
    "created_date":"2015-01-09 11:13:20 PST",
    "last_updated_date":"2015-01-09 12:02:21 PST",
    "first_sent_date":"2015-01-09 11:13:21 PST",
    "last_sent_date":"2015-01-09 11:43:52 PST",
    "payer_view_url":"https://www.sandbox.paypal.com/cgi_bin/webscr?cmd=_pay-inv&viewtype=altview&id=INV2-RV68-2EBS-DTW4-KYHW"
  }
}
````

**Comment by jaypatel512** [comment]
No no. I think this is enough :) Let me ask the internal team developing invoicing api about this. Not sure about the history of Invoicing API. They should be able to figure out why this is happening. 

In the meantime, if you want to get rid of the error, and move ahead, you could add a getter setter for those by copy pasting this code to your "PayPal\Api\Invoice.php" file:

```php
    /**
     * List of payment details for the invoice.
     *
     * @param \PayPal\Api\PaymentDetail[] $payment_details
     *
     * @return $this
     */
    public function setPayments($payment_details)
    {
        $this->payments = $payment_details;
        return $this;
    }

    /**
     * List of payment details for the invoice.
     *
     * @return \PayPal\Api\PaymentDetail[]
     */
    public function getPayments()
    {
        return $this->payments;
    }

```

Let me know if this fixes your issue. In the meantime, I will re-open the issue, and get in touch with internal team about this.

Thanks

**Comment by sebdesign** [comment]
Thank you very much for the workaround and your help. This does fix the error. Let me know if I can be of any help.

Cheers!

**Comment by jaypatel512** [comment]
Thanks for giving us that object json response. Really helped understand the root cause. Its all now on the internal team to help us get this resolved. I have already sent them an email, and I will update you on if I will need any additional information, etc on this. 

Thank you for your understanding.

**Comment by jaypatel512** [comment]
Fixed this Issue on this commit: https://github.com/paypal/PayPal-PHP-SDK/commit/095ab24b626fe698b4be39ca9770968e91723e42

Will be released officially with 1.0.0 version
